examples/silentpayments.c

```diff
@@ -0,0 +1,417 @@
+            size_t n_found_outputs;
+            unsigned int m = 1;
+            struct labels_cache labels_cache;
+
+            for (i = 0; i < N_TX_OUTPUTS; i++) {
+                found_output_ptrs[i] = &found_outputs[i];
+            }
+
+            /* In this contrived example, our label context needs the secp256k1 context because our lookup function
+             * is calling `secp256k1_ec_pubkey_cmp`. In practice, this context can be anything the lookup function needs.
+             */
+            labels_cache.ctx = ctx;
+
+            /* Load Bob's spend public key */
+            ret = secp256k1_ec_pubkey_parse(ctx, &spend_pubkey, bob_spend_pubkey, 33);
+
+            /* Add an entry to the cache. This implies Bob has previously called `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey`
+             * and used the labelled spend pubkey to encode a labelled silent payments address.
+             */
+            ret = secp256k1_silentpayments_recipient_create_label_tweak(ctx,
+                &labels_cache.entries[0].label,
+                labels_cache.entries[0].label_tweak,
+                bob_scan_key,
+                m
+            );
+            assert(ret);
+            labels_cache.entries_used = 1;
+
+            /* Bob collects the data from the transaction inputs and creates a `secp256k1_silentpayments_public_data` object.
+             * He uses this for his own scanning and also serializes the `public_data` object to send to light clients. We will
+             * use this later for Carol, who is scanning as a light client. Note, anyone can create and vend these `public_data`
+             * objecs, i.e. you don't need to be a silent payments wallet, just someone interested in vending this data to light
+             * clients, e.g. a wallet service provider. In our example, Bob is scanning for himself but also sharing this data
+             * with light clients.
+             */
+            ret = secp256k1_silentpayments_recipient_public_data_create(ctx,
+                &public_data,
+                smallest_outpoint,
+                tx_input_ptrs, N_TX_INPUTS,
+                NULL, 0 /* null because no eligible plain pubkey inputs were found in the tx */
+            );
+            assert(ret);
+            /* Save the `public_data` output. This combines the `input_hash` scalar and public key sum by multiplying `input_hash * A_sum`.
+             * The output is then saved as a 33 byte compressed key. Storing it this way saves 32 bytes for the light client because
+             * now it can be send as a 33 byte compressed public key instead of 33 bytes for A_sum and 32 bytes for input_hash.
+             */
+            ret = secp256k1_silentpayments_recipient_public_data_serialize(ctx, light_client_data33, &public_data);
+            assert(ret);
+
+            /* Scan the transaction */
+            n_found_outputs = 0;
+            ret = secp256k1_silentpayments_recipient_scan_outputs(ctx,
```

Comment:
The documentation for `secp256k1_silentpayments_recipient_scan_outputs` needs to be moved below the typedef and struct. At least otherwise VSCode doesn't find it.

Reply:
Nice catch! Will fix and add documentation for the struct.

Reply:
Fixed.